testing/trino-tests/src/test/java/io/trino/execution/TestEventListenerBasic.java

```diff
@@ -623,27 +641,95 @@ public void testOutputColumnsForCreateTableAsSelect()
         QueryCompletedEvent event = getOnlyElement(generatedEvents.getQueryCompletedEvents());
         assertThat(event.getIoMetadata().getOutput().get().getColumns().get())
-                .containsExactly("name", "nationkey");
+                .containsExactly(
+                        new OutputColumnMetadata("name", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "nation", "name"))),
+                        new OutputColumnMetadata("nationkey", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "nation", "nationkey"))));
     }
 
     @Test
     public void testOutputColumnsForCreateTableAsSelectWithAliasedColumn()
             throws Exception
     {
-        runQueryAndWaitForEvents("CREATE TABLE mock.default.create_new_table(aliased_bigint, aliased_varchar) AS SELECT name, nationkey as keynation FROM nation", 2);
+        runQueryAndWaitForEvents("CREATE TABLE mock.default.create_new_table(aliased_varchar, aliased_bigint) AS SELECT concat(name, comment), nationkey as keynation FROM nation", 2);
         QueryCompletedEvent event = getOnlyElement(generatedEvents.getQueryCompletedEvents());
         assertThat(event.getIoMetadata().getOutput().get().getColumns().get())
-                .containsExactly("aliased_bigint", "aliased_varchar");
+                .containsExactly(
+                        new OutputColumnMetadata("aliased_varchar", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "nation", "name"), new ColumnDetail("tpch", "tiny", "nation", "comment"))),
+                        new OutputColumnMetadata("aliased_bigint", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "nation", "nationkey"))));
+    }
+
+    @Test
+    public void testOutputColumnsForCreateTableAsSelectWithAggregation()
+            throws Exception
+    {
+        runQueryAndWaitForEvents("CREATE TABLE mock.default.create_new_table_with_aggregation(aliased_aggregation) AS SELECT sum(totalprice) FROM orders GROUP BY custkey", 2);
+        QueryCompletedEvent event = getOnlyElement(generatedEvents.getQueryCompletedEvents());
+        assertThat(event.getIoMetadata().getOutput().get().getColumns().get())
+                .containsExactly(new OutputColumnMetadata("aliased_aggregation", ImmutableSet.of(new ColumnDetail("tpch", "tiny", "orders", "totalprice"))));
+    }
+
+    @Test
+    public void testOutputColumnsForCreateTableAsSelectWithCountAll()
+            throws Exception
+    {
+        runQueryAndWaitForEvents("CREATE TABLE mock.default.create_new_table_with_aggregation(aliased_aggregation) AS SELECT count(*) FROM orders GROUP BY custkey", 2);
+        QueryCompletedEvent event = getOnlyElement(generatedEvents.getQueryCompletedEvents());
+        assertThat(event.getIoMetadata().getOutput().get().getColumns().get())
+                .containsExactly(new OutputColumnMetadata("aliased_aggregation", ImmutableSet.of()));
+    }
+
+    @Test
+    public void testOutputColumnsForCreateTableAsSelectWithWindowFunction()
```

Comment:
Please add a similar test with `WINDOW` clause instead of in-line window specification.